libs/npm/core/src/components/grid-table/GridTable.tsx

```diff
@@ -184,6 +190,12 @@ export const GridTable = <T extends object>({
     pageSize,
   };
 
+  // The user / system disables a column
+  React.useEffect(() => {
+    if (hiddenColumns) setHiddenColumns(hiddenColumns);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Every time I see this I cringe.  It almost always means we're doing something wrong.  What causes the infinite loop here if you include `setHiddenColumns`?

Reply:
I was trying to debug why `setHiddenColumns` was triggering infinite loop - had no luck yet. Will spend a little more time on it. I know `setState` normally causes a similar issue. Will do some more research

Reply:
Can't figure this one out - some people say ignore the linting warning, others say there are ways around it. This one is specifically hard as `setHiddenColumns` is under the hood with `react-table` 😖